src/services/auth/auth-router.ts

```diff
@@ -14,6 +17,37 @@ for (const key in DeviceRedirects) {
 
 const authRouter = Router();
 
+// Add role to userId by email address (admin only endpoint)
+authRouter.put(
+    "/addRoleByEmail/",
+    RoleChecker([Role.Enum.ADMIN]),
+    async (req, res, next) => {
+        try {
+            const email: string = req.body.email as string;
+            const role: string = req.body.role as string;
+            const user = await Database.ROLES.findOne({ email: email });
+
+            if (!user) {
+                return res.status(StatusCodes.NOT_FOUND).json({
+                    error: "UserNotFound",
+                });
+            }
+
+            const userRoles: string[] = user.roles as string[];
+
+            // Add role if it does not exist
+            if (!userRoles.includes(role)) {
+                userRoles.push(role);
+                await user.save();
+            }
+
+            return res.status(StatusCodes.OK).json(user);
+        } catch (error) {
+            next(error);
+        }
+    }
```

Comment:
instead of find and then push, just do findOneAndUpdate, and add the role to the user directly